README.md

```diff
@@ -557,12 +591,262 @@ end
+- `kwarg(x::SomeType=default_value)`
+  - **it becomes** `x=default_value` (kwarg) in all method definitions, argument will be converted to `SomeType`.
+  - **it means** it's a regular old keyword argument. the value gets converted to the right type if it can be.
+- `typedarg(x::SomeType)`
+  - **becomes** `x::SomeType` in all method signatures
+  - **it means** the argument is required and must be a `SubType` instance.
+- `typedarg(x::SomeType=default_value)`
+  - **results** in two kinds of method signatures, one with `x::SomeType`; the other without `x` in the list of arguments, the constructor then uses the default value in its place.
+- `typedkwarg(x::SomeType=default_value)`
+  - **becomes** `x::SomeType=default_value` (kwarg) in all method definitions.
+  - **means** a keyword argument which must be a `SomeType` instance
+- `curry(x::SomeType)`
+  - **results** in the creation of two kinds of methods. One which has the argument `x::SomeType` in its signature, another that does not have an argument in its place. The latter method returns a lambda that takes x and calls the former method to actually construct the type.
+  - **means** if this argument is missing, then you get back a lambda which you can call with the missing argument. Usually only the last non-keyword argument, if any, is created with curry. This makes `|>` convenient to use in many cases. This argument encourages a free-flowing experimental style. You can more easily write `<a long expression> |> x` than `x(<a long expression>)`, handy while playing with changes, sometimes `|>` reads better.
+
+**Example 13: @api macro usage**
+
+``` julia
+@api border => (Bordered <: Tile) begin
+    arg(side::Side)
+    curry(tile::Tile)
+    kwarg(color::Color=colorant"black")
+    typedkwarg(thickness::Length=1pt)
+end
+```
+
+**Generated definitions**
+
+---------------------------------------------
+
+``` julia
+border(side::Any, tile::Any; color=colorant"black", thickness::Length=1pt) = Bordered(side, convert(Tile, tile), convert(Color, color), thickness)
+border(side::Any; color=colorant"black", thickness::Length=1pt) = tile -> Bordered(side, tile, convert(Color, color), thickness)
+```
+
+---------------------------------------------
+
+The `Tile` argument is the object that will be getting the border in this case. This is a general style in Julia, you construct new tiles to endow some property to an input tile.
+
+The user can call it in two different ways:
+
+``` julia
+border(side, tile, color=my_color, thickness=2pt)
+tile |> border(side, color=my_color, thickness=2pt)
+```
+
+**Example 14: @api macro usage**
+
+``` julia
+@api border => (Bordered <: Tile) begin
+    arg(style::BorderStyle)
+    typedarg(side::Array{Sides}=[left,right,top,bottom])
+    curry(tile::Tile)
+    kwarg(color::Color=colorant"black")
+end
+```
+
+
+**Generated definitions**
+
+---------------------------------------------
```

Comment:
I used the lines to point out the output of the examples, and in this case the result of the @api macro